### PR TITLE
implemented auto-generating unique slugs

### PR DESCRIPTION
**Pull request by Sajal0208**
## What does this PR do?

This PR resolves the issue mentioned in #479

### CODE REVIEW

**Review by a20hek — CHANGES_REQUESTED** [review]
have encountered 2 bugs, can check them out here @Sajal0208 :
https://www.loom.com/share/bfa1af8b19934af19622bc57a0bb49cd?sid=ea5c7b30-bf97-4e4b-8ad6-986dacd68882

**Comment by Sajal0208** [comment]
will fix them right away

File: src/features/listing-builder/components/ListingBuilder/ListingBasic.tsx [hunk 1]
```diff
@@ -80,17 +94,113 @@ export const ListingBasic = ({
...
     subSkills: false,
     skills: false,
     pocSocials: false,
     timeToComplete: false,
   });
 
+  const [isSlugGenerating, setIsSlugGenerating] = useState(false);
+  const [slugErrorMsg, setSlugErrorMsg] = useState('');
   const [isUrlValid, setIsUrlValid] = useState(true);
+  const [shouldSlugGenerate, setShouldSlugGenerate] = useState(false);
```
Comment (a20hek) [inline src/features/listing-builder/components/ListingBuilder/ListingBasic.tsx:107]: there is no need for this
all you need to do is rerender only on title change by adding `title` as a dependancy to the useEffect

**Review by a20hek — CHANGES_REQUESTED** [review]
also, there is no need for two separate endpoints. please do things as necessary

File: src/features/listing-builder/components/ListingBuilder/ListingBasic.tsx (hunk 1 above)
Comment (a20hek) [reply src/features/listing-builder/components/ListingBuilder/ListingBasic.tsx:107]: no, its the flawed getUniqueSlug function. you are trying to rectify it by adding more code

**Comment by PratyushSawan** [comment]
Hi @Sajal0208 and @a20hek,
I just want to give feedback on slug generation. I think we don't have to allow any kind of space/special_charcter in the slug.

What you think @a20hek  on this?

**Comment by a20hek** [comment]
> Hi @Sajal0208 and @a20hek, I just want to give feedback on slug generation. I think we don't have to allow any kind of space/special_charcter in the slug.
> 
> What you think @a20hek on this?

yeah that's correct

**Review by a20hek — APPROVED** [review]

**Pull request merged**